src/Http/Routing/src/Builder/OpenApiDelegateEndpointConventionBuilderExtensions.cs

```diff
@@ -109,15 +109,15 @@ public static DelegateEndpointConventionBuilder ExcludeFromDescription(this Dele
         /// </summary>
         /// <param name="builder">The <see cref="DelegateEndpointConventionBuilder"/>.</param>
         /// <param name="statusCode">The response status code. Defaults to StatusCodes.Status400BadRequest.</param>
-        /// <param name="contentType">The response content type. Defaults to "application/validationproblem+json".</param>
+        /// <param name="contentType">The response content type. Defaults to "application/problem+json".</param>
         /// <returns>A <see cref="DelegateEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
         public static DelegateEndpointConventionBuilder ProducesValidationProblem(this DelegateEndpointConventionBuilder builder,
             int statusCode = StatusCodes.Status400BadRequest,
             string? contentType = null)
         {
             if (string.IsNullOrEmpty(contentType))
             {
-                contentType = "application/validationproblem+json";
+                contentType = "application/problem+json";
```

Comment:
Is MVC also affected by this? Could we also fix it while we're addressing this?

Reply:
AFAICT, MVC isn't affected by this as far as I can tell. The `ControllerBase.ValidationProblem` method uses the `application/problem+json` content type and the `ProducesResponseType` attribute expects that users pass the content type for when they specify a `typeof(ValidationProblemDetails)` hence the fix for the test case below.